### PR TITLE
Space picker port, button icon props, and font weight variables

### PR DESCRIPTION
**Pull request by 1egoman**
- Port over the space picker from the dashboard
- Add a new `leftIcon` and `rightIcon` props to the button so that an icon can be rendered on the left and right of a button. This works similarly to the existing `leftIcon` prop on the `InputBox`, and matches the styling of the most up to date sketch density ui button placement:
Screen Shot 2020-06-12 at 11 25 29 AM
Screen Shot 2020-06-12 at 11 25 35 AM
- Adds some new font variables that define font weights. I pushed for this a while back when the the colors were redone, but it was forgotten about. I don't think this is controversial though.

**NOTE: The package version still needs to be updated.**

### CODE REVIEW

File: src/space-picker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
+import React, { useContext, useState, Component, Fragment } from 'react';
```
Comment (1egoman) [inline src/space-picker/index.tsx:1]: Note: this was a direct port from the dashboard, I didn't change anything.

File: src/button/index.tsx [hunk 2]
```diff
@@ -54,6 +58,29 @@ const Button = React.forwardRef<HTMLElement, ButtonProps>((props, ref) => {
...
+    contents = (
+      <Fragment>
+        <div className={classnames(styles.icon, styles.left)}>
+          {leftIcon}
+        </div>
+        {contents}
+      </Fragment>
+    );
+  }
+  if (rightIcon) {
```
Comment (guscost) [inline src/button/index.tsx:73]: Weird next-line if is non-javascripty

File: src/input-box/styles.module.scss [hunk 3]
```diff
@@ -58,6 +58,21 @@
   }
 }
 
+.inputBoxContainsRightIcon { padding-right: 16px; }
+.inputBoxContainsRightIcon input { margin-right: 10px; }
```
Comment (guscost) [inline src/input-box/styles.module.scss:62]: Weird, why the margin???

Comment (guscost) [reply src/input-box/styles.module.scss:62]: Not a dealbreaker though, this is just continuing with what was there.

**Pull request merged**